api.py

```diff
@@ -1,11 +1,97 @@
-from flask import Flask
+import json
+import os
+import urllib
+from json import JSONDecodeError
 
-from app.views.status import status_blueprint
-from app.views.validate import validate_blueprint
+import requests
+import uvicorn
+from fastapi import Body, FastAPI, Response
+from requests import RequestException
+from structlog import get_logger
+
+from app.validators.questionnaire_validator import QuestionnaireValidator
+
+app = FastAPI()
+
+
+@app.get("/status")
+async def status():
+    return Response(status_code=200)
+
+
+logger = get_logger()
+
+AJV_HOST = os.getenv("AJV_HOST", "localhost")
+
+AJV_VALIDATOR_URL = f"http://{AJV_HOST}:5002/validate"
+
+
+@app.post("/validate")
+async def validate_schema_request_body(payload=Body(None)):
+    logger.info("Validating schema")
+    return await validate_schema(payload)
+
+
+@app.get("/validate")
+async def validate_schema_from_url(url: str = ""):
+    if url:
+        logger.info("Validating schema from URL", url=url)
+        try:
+            with urllib.request.urlopen(url) as opened_url:
+                return await validate_schema(data=opened_url.read().decode())
+        except urllib.error.URLError:
+            return (
+                Response(
+                    status_code=404,
+                    content=f"Could not load schema at URL [{url}]",
+                ),
+            )
+
+
+async def validate_schema(data):
+    json_to_validate = None
+    if data:
+        if isinstance(data, str):
+            try:
+                json_to_validate = json.loads(data)
+            except JSONDecodeError:
+                logger.info("Could not parse JSON", status=400)
+                return Response(status_code=400, content="Could not parse JSON")
+        elif isinstance(data, dict):
+            json_to_validate = data
+
+    response = {}
+    try:
+        ajv_response = requests.post(
+            AJV_VALIDATOR_URL, json=json_to_validate, timeout=10
+        )
+        ajv_response_dict = ajv_response.json()
+
+        if ajv_response_dict:
+            response["errors"] = ajv_response_dict["errors"]
+            logger.info("Schema validator returned errors", status=400)
+            return response, 400
+
+    except RequestException:
+        logger.info("AJV Schema validator service unavailable")
+        return json.dumps(obj={}, error="AJV Schema validator service unavailable")
+
+    validator = QuestionnaireValidator(json_to_validate)
+    validator.validate()
+
+    if len(validator.errors) > 0:
```

Comment:
V minor suggestion - This was already here before, but could do `if validator.errors` instead?

Reply:
Added.